src/aiida/tools/dumping/processes.py

```diff
@@ -285,7 +285,15 @@ def _dump_calculation(
         # Dump the node_inputs
         if self.include_inputs:
             input_links = calculation_node.base.links.get_incoming(link_type=LinkType.INPUT_CALC)
-            self._dump_calculation_io(parent_path=output_path / io_dump_mapping.inputs, link_triples=input_links)
+            all_input_nodes = input_links.all_nodes()
+            all_have_repositories = all([hasattr(node.base, 'repository') for node in all_input_nodes])
+            if all_have_repositories:
+                non_empty_repository = any([len(node.base.repository.list_objects()) > 0 for node in all_input_nodes])
+                if non_empty_repository:
+                    self._dump_calculation_io(
+                        parent_path=output_path / io_dump_mapping.inputs,
+                        link_triples=input_links,
+                    )
```

Comment:
```suggestion
all_input_nodes = input_links.all_nodes()
if (
    all(hasattr(node.base, 'repository') for node in all_input_nodes)
    and any(len(node.base.repository.list_objects()) for node in all_input_nodes)
):
    self._dump_calculation_io(
        parent_path=output_path / io_dump_mapping.inputs,
        link_triples=input_links,
    )
```

I think it can be simplified by merging two if?